runtime/compiler/control/CompilationThread.cpp

```diff
@@ -1766,6 +1766,12 @@ bool TR::CompilationInfo::canProcessLowPriorityRequest()
    if (getMethodQueueSize() != 0)
       return false;
 
+#if defined(J9VM_OPT_JITSERVER)
+   // If the first LPQ entry is REASON_SERVER_UNAVAILABLE, only attempt compilation if server is available by now
+   if (getLowPriorityCompQueue().getFirstLPQRequest()->_reqFromSecondaryQueue == TR_MethodToBeCompiled::REASON_SERVER_UNAVAILABLE)
```

Comment:
We may need the compilation queue monitor here because by the time we call `getLowPriorityCompQueue().getFirstLPQRequest()->_reqFromSecondaryQueue` another thread could have made the queue empty (unless the caller already has the monitor)

Reply:
The monitor is already acquired everywhere `canProcessLowPriorityRequest` is called, since it's only used around the code that already modifies compilation queues.